docs/src/components/Layout/index.tsx

```diff
@@ -54,6 +54,7 @@ export default function Page({
         }))
       );
     });
+    updateHeaders(); // with static rendering the mutation observer is no longer triggered on initial page view because the content has already been rendered
```

Comment:
This line was added because we need to generate the initial page headings, this was being handled previously by the MutationObserver after the dynamic content was loaded into the page, but now that the dynamic content has been removed the observer is no longer triggering on the initial load.

Reply:
Do we still need `observer`?

Reply:
Is it problematic to call it here and from `observer`?

Reply:
I don't think its problematic calling `updateHeaders` in both places because this one is being called with `useLayoutEffect` at the same time as the observer is being created but the observer doesn't trigger until the page content is changed at which time updateHeaders is called again.